keras/engine/topology.py

```diff
@@ -1943,7 +1943,15 @@ def losses(self):
                 losses += layer.get_losses_for(None)
         # Add any potential unconditional model-level loss.
         losses += self.get_losses_for(None)
-        return losses
+
+        losses_without_duplicates = []
```

Comment:
- Wouldn't `list(set(losses))` do the same thing more efficiently?
- Wouldn't testing for object identity (`is`) be an issue for losses that are integers or floats? (exact same behavior as use of `id`).

Reply:
I agree with you. How do you suggest that we test for object identity? I could make a pseudo static variable in add_weight, but I'm not sure it's cleaner than using is or id.

Reply:
We could do:

```python
unique_tensors = list(set([x for x in losses if not isinstance(x, (float, int))]))
non_tensors = [x for x in losses if isinstance(x, (float, int))]
losses = unique_tensors + non_tensors
```

What do you think?